### PR TITLE
allow overriding protocol contracts addresses for deployment

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
# use custom PoCo instance in the deployment

add the possibility to deploy contracts and subgraph with custom protocol contract addresses

the following env overrides the default (bellecour) addresses

- POCO_ADDRESS
- DATASET_REGISTRY_ADDRESS
- APP_REGISTRY_ADDRESS

misc

- internal variables have been renamed

### CODE REVIEW

File: packages/sharing-smart-contract/hardhat.config.cjs [hunk 1]
```diff
@@ -40,6 +40,15 @@ module.exports = {
       url: 'https://bellecour.iex.ec',
       accounts: WALLET_PRIVATE_KEY ? [WALLET_PRIVATE_KEY] : [],
     },
+    // poco-chain native config
+    'dev-native': {
+      chainId: 65535,
```
Comment (Le-Caignec) [inline packages/sharing-smart-contract/hardhat.config.cjs:45]: where does this chainId come from ?

**Review by Le-Caignec — APPROVED** [review]
LGTM
May be it could be nice to start building an `env.template` file for both sharing-smart-contract and smart-contract

File: packages/sharing-smart-contract/hardhat.config.cjs (hunk 1 above)
Comment (PierreJeanjacquot) [reply packages/sharing-smart-contract/hardhat.config.cjs:45]: internal poco-chain repo => https://github.com/iExecBlockchainComputing/poco-chain

**Comment by PierreJeanjacquot** [comment]
@Le-Caignec  added env templates [f5143fa](https://github.com/iExecBlockchainComputing/dataprotector-sdk/pull/316/commits/f5143fae0f7001f47fbfa2b243700cca0cac6b8f)

**Pull request merged**